frontend/src/Components/ProfilePage/ProfilePage.jsx

```diff
@@ -0,0 +1,231 @@
+        alert("Profile updated successfully");
+      } else {
+        console.error("Failed to update profile");
+        alert("Failed to update profile");
+      }
+    } catch (error) {
+      console.error("Error updating profile:", error);
+      alert("Failed to update profile");
+    }
+  };
+
+  return (
+    <div className="profile-page">
+      <button
+        className={`sidebar-button-modern ${isSidebarOpen ? "hidden" : "visible"}`}
+        onClick={() => setIsSidebarOpen(true)}
+      >
+        <i className="fas fa-bars"></i>
+      </button>
+      <div className="profile-card">
+        <div className="profile-picture">
+          <img
+            src={
+              profile.profilePicture
+                ? `http://localhost:3000/${profile.profilePicture}`
+                : "http://localhost:3000/default-profile-photo.jpeg"
+            }
+            alt="Profile"
+          />
+        </div>
+        <div className="profile-info">
+          <h2>{profile.name}</h2>
+          <p>Location: {profile.location || "N/A"}</p>
+          <button onClick={() => setIsModalOpen(true)}>
+            <i className="fas fa-edit"></i>
+          </button>
+        </div>
+        <Tabs>
+          <TabList>
+            <Tab>Bio</Tab>
+            <Tab>Pinned Services</Tab>
+            <Tab>Reviews</Tab>
+          </TabList>
+
+          <TabPanel>
+            <div className="profile-bio">
+              <h3>Bio</h3>
+              <p>{profile.bio || "N/A"}</p>
+              <h3>Interests</h3>
+              <p>{profile.interests || "N/A"}</p>
+            </div>
+          </TabPanel>
+          <TabPanel>
+            <div className="pinned-services">
+              <h3>Pinned Services</h3>
+              <div className="services-container">
+                {pinnedServices.length > 0 ? (
+                  pinnedServices.map((service, index) => (
+                    <div key={index} className="service-card">
+                      <h4>{service.name}</h4>
+                      <p>{service.description}</p>
+                    </div>
+                  ))
+                ) : (
+                  <img
+                    src="default-pinned-service-image.jpeg"
+                    alt="Default Pinned Service"
+                  />
+                )}
+              </div>
+            </div>
+          </TabPanel>
+          <TabPanel>
+            <div className="profile-reviews">
+              <h3>Reviews</h3>
+              <p>Here will be the user's reviews...</p>
+              {/* Implement fetching and displaying user reviews */}
+            </div>
+          </TabPanel>
+        </Tabs>
+      </div>
+
+      {isModalOpen && (
+        <Modal onClose={() => setIsModalOpen(false)}>
+          <form onSubmit={handleSubmit} className="profile-form">
+            <div className="form-group">
+              <label>Profile Picture</label>
+              <input type="file" onChange={handleFileChange} />
```

Comment:
Should this file input accept any file MIME type or only images?

Also as a sanity-check does this input accept multiple files or just a single file?

Reply:
The file input should only accept image files since it's meant for a profile picture. To ensure this, I/ve added an accept attribute which is set to image/*.
Regarding the number of files, this input currently accepts only a single file.